### PR TITLE
perf(oauth): draft oauth/register action

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/component/driver/oauth/utility/utility.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+export async function getandValidateOauthInfoFromSpec(
+  args: CreateOauthArgs,
+  context: DriverContext,
+  actionName: string
+): Promise<OauthInfo> {
+  const absolutePath = getAbsolutePath(args.apiSpecPath, context.projectPath);
+  const parser = new SpecParser(absolutePath, {
+    allowBearerTokenAuth: isApiKeyEnabled(), // Currently, API key auth support is actually bearer token auth
+    allowMultipleParameters: isMultipleParametersEnabled(),
```
Comment (SLdragon) [inline packages/fx-core/src/component/driver/oauth/utility/utility.ts:39]: isApiKeyEnabled() feature flag is used for SME and already been set to true, I think we should use another one for copilot.

Comment (KennethBWSong) [reply packages/fx-core/src/component/driver/oauth/utility/utility.ts]: Updated, thanks!

**Review by SLdragon — APPROVED** [review]

**Review by nliu-ms — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -852,5 +852,13 @@
   "driver.apiKey.error.domainInvalid": "Domain is invalid. Domain for API key should follow: 1. Max %d domain per API key. 2. Use comma to separate domains",
   "driver.apiKey.error.failedToGetDomain": "Failed to get domain from API specification. Please make sure your API specification is valid.",
   "driver.apiKey.log.successCreateApiKey": "Created API key with id %s",
-  "driver.apiKey.log.failedExecuteDriver": "Unable to execute action %s. Error message: %s"
+  "driver.apiKey.log.failedExecuteDriver": "Unable to execute action %s. Error message: %s",
+  "driver.oauth.description.create": "Create an Oauth registration on Developer Portal for authentication in Open API spec.",
+  "driver.oauth.title.create": "Creating Oauth registration...",
+  "driver.oauth.log.skipCreateOauth": "Environment variable %s exists. Skip creating API key.",
+  "driver.oauth.log.oauthNotFound": "Environment variable %s exists but failed to retrieve Oauth registration from Developer Portal. Check manually if it exists.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:859]: Revised: Environment variable %s exists but unable to retrieve 0auth registration from Developer Portal. Check manually if it exists.

Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json]: Updated, thanks！

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -852,5 +852,13 @@
...
+  "driver.oauth.error.nameTooLong": "The name for Oauth is too long. The maximum length is 128.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:860]: Revised: The 0auth name is too long. The maximum character length is 128.

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -852,5 +852,13 @@
...
+  "driver.oauth.log.successCreateOauth": "Oauth registration creted successfully with id %s!",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:861]: Spelling error - change it to 'created'

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -852,5 +852,13 @@
...
+  "driver.oauth.error.domainInvalid": "Domain is invalid. Max %d domain per Oauth registration.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:862]: Revised: Maximum %d domains allowed per 0auth registration.
Logic: 'Domain is invalid' is irrelevant and not necessary. We only need to convey how many domains are allowed per registration.

File: packages/fx-core/resource/package.nls.json [hunk 6]
```diff
@@ -852,5 +852,13 @@
...
+  "driver.apiKey.error.oauthAuthInfoInvalid": "Failed to parse oauth2 authScheme from spec. Please make sure your API specification is valid."
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:863]: Revised: Unable to parse oauth2 authScheme from spec. Make sure your API specification is valid.

File: packages/fx-core/resource/package.nls.json (hunk 5 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json]: Update, thanks!

File: packages/fx-core/resource/package.nls.json (hunk 4 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json]: Update, thanks!

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json]: Update, thanks!

File: packages/fx-core/src/component/driver/oauth/utility/utility.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+export async function getandValidateOauthInfoFromSpec(
+  args: CreateOauthArgs,
+  context: DriverContext,
+  actionName: string
+): Promise<OauthInfo> {
+  const absolutePath = getAbsolutePath(args.apiSpecPath, context.projectPath);
+  const parser = new SpecParser(absolutePath, {
+    allowAPIKeyAuth: false,
+    allowBearerTokenAuth: isCopilotAuthEnabled(),
```
Comment (jayzhang) [inline packages/fx-core/src/component/driver/oauth/utility/utility.ts:39]: suggest to use the new API to access feature flag: `featureFlagManager.getBooleanValue(FeatureFlags.CopilotAuth)`

**Review by SLdragon — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/component/driver/oauth/utility/utility.ts (hunk 7 above)
Comment (KennethBWSong) [reply packages/fx-core/src/component/driver/oauth/utility/utility.ts:39]: Will fix in another pr. Thanks!

**Review by supkasar — APPROVED** [review]

**Review by wenytang-ms — APPROVED** [review]

**Pull request merged**